applications/schema/schema/events/switch_controllers.schema.json

```diff
@@ -0,0 +1,66 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Switch Controllers",
+  "description": "Start or stop controllers on a hardware interface",
+  "oneOf": [
+    {
+      "type": "array",
+      "minItems": 1,
+      "uniqueItems": true,
+      "items": {
+        "$ref": "#/$defs/switch_controllers_object"
+      }
+    },
+    {
+      "$ref": "#/$defs/switch_controllers_object"
+    }
+  ],
+  "$defs": {
+    "switch_controllers_object": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "interface": {
+          "$ref": "identifiers/hardware.schema.json"
+        },
+        "start": {
+          "title": "Start Controllers",
+          "description": "The name(s) of the controller(s) to start",
+          "additionalProperties": false,
+          "oneOf": [
+            {
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "$ref": "identifiers/controller.schema.json#/properties/controller"
+              }
+            },
+            {
+              "$ref": "identifiers/controller.schema.json#/properties/controller"
+            }
+          ]
+        },
+        "stop": {
+          "title": "Stop Controllers",
+          "description": "The name(s) of the controller(s) to stop",
+          "additionalProperties": false,
+          "oneOf": [
+            {
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "$ref": "identifiers/controller.schema.json#/properties/controller"
+              }
+            },
+            {
+              "$ref": "identifiers/controller.schema.json#/properties/controller"
+            }
+          ]
+        }
+      },
+      "required": [
+        "interface"
```

Comment:
Here, would it be possible to do something like
```
allOf
  required [interface]
  anyOf
     required [ start]
     required [stop]
```
to make sure at least start or stop is there?

Reply:
Yes I realise I took that out from your previous version. I will check again but I think it was causing problems with the validation ("the schema validates to multiple variants").

Reply:
Ok you are correct, the `allOf` container works and solves the multiple variant problem 👍